### ISSUE TITLE
running xrdp as non root

### ISSUE THREAD

**Issue by cyberb** [neutrinolabs/xrdp#807]
I have no root access and would like to use xrdp.
I have compiled xrdp and xorg.

Everything was compiled and installed with prefix.

Issues:

1. Unable to authenticate as I have no access to /etc/pam.d. Had to comment authentication in the code and accept user as is.

2. Unable to run Xorg as I have no access to /etc. Xorg then tried to use default config and complains about various drivers. Cannot find any solution yet.

Is it possible?

**Comment by metalefty** [comment]
I think files in /etc/pam.d should be user readable regardless of xrdp's running privilege.

Talking about 2, install xrdp's xorg.conf to another place and replace the path in sesman.ini with full path to there.

```
[Xorg]
param=Xorg
param=-config
param=/full/path/to/xrdp/xorg.conf
param=-noreset
param=-nolisten
param=tcp
```

**Comment by cyberb** [comment]
Files in /etc/pam.d are not valid and I cannot change them.

Xorg does not allow to supply absolute path when running non root. I have my own build. I guess I will have to these it as well.

**Comment by geiseri** [comment]
With Xwrapper this should work.  I think the official debian packages in fact cause xrdp to run as the xrdp user.

**Comment by volth** [comment]
Almost all distros has patches for this (e.g. Debian: http://sources.debian.net/src/xrdp/0.9.1-9/debian/patches/)

Basically it is only replace with replace ```g_sck_listen(dev->listen_sck);``` -> ```g_sck_listen(dev->listen_sck); g_chmod_hex(dev->uds_data, 0x0660);``` in ```module/rdpClientCon.c```

**Comment by metalefty** [comment]
Maybe it is actually a different issue from "running xrdp as non root".
xrdp should work without root privilege because Debian already does it.

However, the reporter says "Files in /etc/pam.d are not valid and I cannot change them".
It is another problem. The reporter seems doesn't have root privilege.
xrdp can be run without root privilege but requires root privileges to install.

The reporter's real issue looks "install xrdp as non root" to me.

**Comment by cyberb** [comment]
You are right I have no root at all on the box.
As a result all those problems.
I had to disable pam in the code.

Also as a subsequent issue I had to patch all compile config variables to be environment variables as different users have access to different dirs.

As a result I was able.to compile a self-contained relocatable package with no external deps which can be run from any directory by non root users.
